scripts/install_third_party.py

```diff
@@ -80,7 +93,32 @@
 """)
 
 
-def download_files(source_url_root, target_dir, source_filenames):
+class DependencyDict(TypedDict, total=False):
```

Comment:
Using `total=False` since some of the fields in this dict is optional/Not required. By optional, I mean there are possibilities that some fields can be present or not. In some cases, either one of the 2 fields should be present. However, we do have validation for this in code over [here](https://github.com/oppia/oppia/blob/d51789394468f64937215befaa5dfa5baf535759/scripts/install_third_party.py#L271)

Reply:
I am not sure if we could use cast since the optional key condition is, a dict can have either field A or field B
Example over here https://github.com/oppia/oppia/blob/d51789394468f64937215befaa5dfa5baf535759/scripts/install_third_party.py#L73

where the dict can have  either rootDir or rootDirPrefix

Reply:
Ok, I have also tried the approach of NotRequired ([LINK](https://peps.python.org/pep-0655/#usage-in-python-3-11)), but it seems to me that NotRequired is not working here because of The older version of MyPy. So, I think a comment should be added here at least about why it is necessary.